.github/workflows/linter.yml

```diff
@@ -10,6 +10,6 @@ jobs:
         with:
           go-version: '1.24.x'
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v6
+        uses: golangci/golangci-lint-action@v7
```

Comment:
It seems like we are doubling up on checks now so maybe do `[push, pull_request]` -> `[pull_request]` for `on:`

Reply:
[Documentation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request) for reference suggests it will run on opened, synchronize, or reopened.
I *think* synchronize happens whenever you push.

However, I don't think I'd want to lose the implications of running these on push *outside* of PRs, because I use that quite a bit on my own refs.